lib/src/import_path_base.dart

```diff
@@ -0,0 +1,283 @@
+// Copyright (c) 2020, Google Inc. Please see the AUTHORS file for details.
+// All rights reserved. Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:analyzer/dart/analysis/utilities.dart';
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:ascii_art_tree/ascii_art_tree.dart';
+import 'package:package_config/package_config.dart';
+import 'package:path/path.dart' as p;
+
+/// An Import Path search tool.
+class ImportPath {
+  /// The entry point to start the search.
+  final Uri from;
+
+  /// The import to find. Can be an [Uri] or a [RegExp].
+  dynamic importToFind;
```

Comment:
Actually I think it would be worth creating an abstraction here, something like an `ImportFinder` class (or function type). Then you can have Uri and RegExp implementations of that, and we can have a proper type for this field both here and in the actual class, and avoid the repeated `is` checks.